### ISSUE TITLE
False negative: ComparingUnrelatedTypes

### ISSUE THREAD

**Issue by rtfpessoa** [sksamuel/scapegoat#189]
Seems like ComparingUnrelatedTypes is not detecting this case. 
It might be some mistake I made configuring, or scapegoat is not catching the [wrong comparison](https://github.com/rtfpessoa/scapegoat-unrelatedtypescomparision/blob/master/src/main/scala/xyz/rtfpessoa/Example.scala#L20).

Any ideas?

**Comment by xplosunn** [comment]
Turns out this is due to the comparison inspections not being applied in the case of != comparisons. I made a PR.

**Comment by sksamuel** [comment]
The PR is merged now, so closing this issue.

**Comment by rtfpessoa** [comment]
@sksamuel any estimation for a release date?

**Comment by sksamuel** [comment]
I've just tried doing one, but I cannot get the project to build locally.

```
com.sksamuel.scapegoat.inspections.collections.NegationIsEmptyTest *** ABORTED ***
[info]   java.io.FileNotFoundException: Could not locate SBT compile directory for plugin files [./target/scala-2.11./classes]
[info]   at com.sksamuel.scapegoat.PluginRunner$class.sbtCompileDir(PluginRunner.scala:79)
[info]   at com.sksamuel.scapegoat.inspections.collections.NegationIsEmptyTest.sbtCompileDir(NegationIsEmptyTest.scala:7)
[info]   at com.sksamuel.scapegoat.PluginRunner$class.$init$(PluginRunner.scala:17)
[info]   at com.sksamuel.scapegoat.inspections.collections.NegationIsEmptyTest.<init>(NegationIsEmptyTest.scala:7)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[info]   at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[info]   at java.lang.Class.newInstance(Class.java:442)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:435)
```

Can anyone else do `sbt +test` ?

**Comment by rtfpessoa** [comment]
After testing I got the same error multiple times.

**Comment by rtfpessoa** [comment]
Travis seems to be doing only `sbt test` and not `sbt +test` so this might be happening for a while.

**Comment by sksamuel** [comment]
It obviously worked when I did the last release in b2aad17 so I'll bisect and find out which one broke it.

**Comment by rtfpessoa** [comment]
Ok, that make sense.

**Comment by rtfpessoa** [comment]
I think the problem is here https://github.com/sksamuel/scapegoat/blob/master/src/test/scala/com/sksamuel/scapegoat/PluginRunner.scala#L15. In Scala 2.11 the minor version is two numbers and this code seems to be failing.

I would recommend something in the lines of `val shortScalaVersion = scalaVersion.split('.').dropRight(1).mkString(".")`. I tested this locally and works fine.

I did #191 if you want to check.
